src/solc_0.7/MultiGiveaway/MultiGiveawayWithERC20.sol

```diff
@@ -0,0 +1,110 @@
+//SPDX-License-Identifier: MIT
+pragma solidity 0.7.5;
+
+import "@openzeppelin/contracts/token/ERC721/IERC721Receiver.sol";
+import "@openzeppelin/contracts/token/ERC1155/IERC1155Receiver.sol";
+import "./ClaimERC721AndERC1155WithERC20.sol";
+import "../common/BaseWithStorage/WithAdmin.sol";
+
+/// @title MultiGiveaway contract.
+/// @notice This contract manages multiple ERC721,ERC1155 and ERC20 claims.
+contract MultiGiveawayWithERC20 is WithAdmin, ClaimERC721AndERC1155WithERC20 {
```

Comment:
This could be the only contract to focus on.
Unless there are some gas consideration for having ERC focus contract

If we focus on that single contract, we could make a version of it with multi merkle tree so that we can add claims without requiring a new deployment